Documentation/EngineeringPlanLinuxDistroRepos.md

```diff
@@ -0,0 +1,132 @@
+3. Eliminate prebuilt binaries. (https://github.com/dotnet/source-build/issues/753)
+4. Work with DotNet SIG to get packaging ready.
+
+As stated in the requirements, work will be done in the `release/2.1` branch of source-build.  Any changes required to source repos will follow this process:
+1. Include changes in source-build as patches to continue moving work forward.
+2. Write up issue in target repo to include change in `release/2.1` branch and port to `release/2.2` and `master` branches.
+3. Update source-build to move to updated code and remove patch.
+
+Source-build changes will be ported to `release/2.2` and `master` on a regular basis to keep those branches up to date.  The desire is to have `release/2.2` and `master` updated and ready to be included in other distros when they are ready to be released.
+
+### Toolset Bootstrapping ##
+This work requires setting up scripts to build the .NET Core toolset from source and then using the result to build source-build.  This is to define the process and to ensure that it works consistently by adding CI.  It also will require working out issues that we've encountered building source-build with the same version that is being built.  See https://github.com/dotnet/source-build/issues/606.  This will also track work with source-building buildTools and the eventual move to Arcade, if it is ported to the `release/2.1` branch for any repos.
+
+### Runtime / SDK Split ###
+The main part of this work is to split and isolate the runtime and sdk builds into separate directories with separate build scripts in source-build.  The result of the runtime build should be input to the sdk build.  Each should be able to be built independently.
+
+### Prebuilt Binary Elimination ###
+
+There are 6 categories of prebuilts to eliminate:
+1. Prebuilts that end up in the built product. - These must be eliminated and a process must be put into place to detect and track these to make sure they don't happen again.
+2. Prebuilts loaded in unneccessary functionality - These are prebuilts that are loaded by a repo's source even though the functionality for the source is not included in source-build.  The main group of these are test assemblies, since source-build does not build tests.
+3. Prebuilts that have a version built with source-build - The source-build process should be modified to get repos to use the source-built version of the package.  Many of these are older
+4. Prebuilts containing reference assemblies - These prebuilts can be delivered as IL-only code and re-compiled as reference assemblies.
+5. First party prebuilts - If there are dependencies on 1st party packages that are not built as part of source-build, we should investigate including them in source-build if possible or eliminating the dependency.
+6. Third party prebuilts - Steps should be taken to remove any dependencies on 3rd party packages.
```

Comment:
Another possible way to deal with 5 and 6 is to package them up as separate (distro) packages. When building for a distro, it should be perfectly fine to use a dependency that's available in the distribution. This is the case for native code already (clang, lldb, lttng, openssl). Theoretically, we should be able to do the same with .NET packages as well. We could then have a dependency on the already built packages.